barretenberg/cpp/src/barretenberg/dsl/acir_format/acir_to_constraint_buf.hpp

```diff
@@ -10,28 +10,116 @@
 
 namespace acir_format {
 
+/// ========= HELPERS ========= ///
+
+/// The functions below are helpers for converting data between ACIR representations and Barretenberg's internal
+/// representations.
+
+uint256_t from_be_bytes(std::vector<uint8_t> const& bytes);
+
+/**
+ * @brief Parse an Acir::FunctionInput (which can either be a witness or a constant) into a WitnessOrConstant.
+ */
+WitnessOrConstant<bb::fr> parse_input(Acir::FunctionInput input);
+
+/**
+ * @brief Extract the witness index from an Acir::FunctionInput representing a witness.
+ *
+ * @note The function asserts that the input is indeed a witness variant.
+ */
+uint32_t get_witness_from_function_input(Acir::FunctionInput input);
+
+/// ========= BYTES TO BARRETENBERG'S REPRESENTATION  ========= ///
+
+/// The functions below handle the transition from serialized ACIR formats (msgpack and bincode), which is the output of
+/// compiling a Noir program, to Barretenberg's internal formats.
+///
+/// The flow is as follows:
+/// - A buffer of bytes is deserialised according to either msgpack or bincode into an Acir::Circuit, which is just the
+///   representation of a function in terms of Acir::Opcodes, Acir::Witness, Acir::PublicInputs. As of now (Nov 2025)
+///   only bincode is supported.
+/// - The Acir::Circuit is transformed into AcirFormat, which is Barretenberg's internal representation of a circuit,
+///   in terms of constraints that have to be added to the Builder
+/// - A buffer of bytes is deserialised into a WitnessVector, which is the list of witness values used by the prover to
```

Comment:
Being a little pedantic here but it might be clearer here to say "the list of witness values known at the time of noir program execution" or something like that. People have mistakenly thought in the past that the witness we get from noir is the entire witness for the bberg representation of the circuit which is of course not the case